### ISSUE TITLE
Water Leak Sensor H5054 - Not reporting leaks or correct battery status

### ISSUE THREAD

**Issue by schwaggy** [bwp91/homebridge-govee#733]
What issue do you have? Please be as thorough and explicit as possible.

The H5054 Leak sensors do not send leak status to HomeKit, therefore no Critical Alert is generated when water is detected.
The native Govee app has the correct leak info and sends proper push notifications (and email) from Govee.

Also, battery info from the plug-in is incorrect showing all sensors 0%

Details of your setup.

    Do you use (1) Homebridge UI-X (2) Homebridge CLI or (3) HOOBS?

    Homebridge

    Which version of Homebridge/HOOBS do you have?

    1.7.0

    Which platform do you run Homebridge/HOOBS on (e.g. Raspberry Pi/Windows/HOOBS Box)? Please also mention your version of Node.js/NPM if known.

    MacOS Monterey 12.7.2

    Node 20.11.0

    Which version of this plugin (homebridge-govee) do you have? Has the issue started since upgrading from a previous version?

    10.3.0, 10.4.0

 problem started with 10.3.0 - reverting to 10.2.0 works

    Which Govee devices do you have that are causing issues? Please include product models if applicable.

    H5054 Water Leak Sensors

Please paste any relevant logs below.

No obvious errors in the logs. Note I have 16 leak sensor devices but edited the log to only show one of them.

[1/21/2024, 8:25:51 PM] [Govee] Launched child bridge with PID 48901
[1/21/2024, 8:25:51 PM] Registering platform 'homebridge-govee.Govee'
[1/21/2024, 8:25:51 PM] [Govee] Loaded homebridge-govee v10.3.0 child bridge successfully
[1/21/2024, 8:25:51 PM] Loaded 16 cached accessories from cachedAccessories.0E65E4382F51.
[1/21/2024, 8:25:51 PM] [Govee] Initialising plugin v10.3.0 | System darwin | Node v20.11.0 | HB v1.7.0 | HAPNodeJS v0.11.1...
[1/21/2024, 8:25:51 PM] [Govee] Plugin initialised. Setting up accessories....
[1/21/2024, 8:25:51 PM] Homebridge v1.7.0 (HAP v0.11.1) (Govee) is running on port 48354.
[1/21/2024, 8:25:53 PM] [Govee] [LAN] client enabled and found 0 device(s).
[1/21/2024, 8:25:54 PM] [Govee] [HTTP] client enabled and found 16 device(s).
[1/21/2024, 8:25:54 PM] [Govee] [AWS] disabling client as disabled in config.
[1/21/2024, 8:25:54 PM] [Govee] [BLE] disabling client as disabled in config.
[1/21/2024, 8:25:54 PM] [Govee] [Guest Toilet] initialising with options {}.
[1/21/2024, 8:25:54 PM] [Govee] [Guest Toilet] initialised with id [42:F5:20:38:CB:F6:00:18:C3] [H5054].

**Comment by RJ1002** [comment]
same issue. I have 5 H5054. 
my homebridge is running on an unraid server as a docker.
node.js version: v20.11.1
homebridge-govee: v10.4.0

**Comment by SJNix** [comment]
Same issue with me for v10.4.0 -  homebridge on Synology DSM. node.js v20.9.0.

**Comment by schwaggy** [comment]
Still not working
Plug-In 10.5.0
Node 20.12.0

**Comment by lsliddar** [comment]
I am having the same issue:

Homebridge on Synology DSM
Homebridge v1.8.0
Plugin v10.6.0
Nodejs v18.18.2

Maybe something strange with 'initialized with option {}'?? Happy to test in more detail to get this resolved.

**Comment by mynameisbobwehadababyitsaboy** [comment]
Signed up for Github just to comment on here and say this issue is impacting me also. Had a water sensor go off last night, and to my surprise never saw anything come through Homekit or Homebridge. 

- Homebridge on Raspberry Pi (Raspbian GNU/Linux Bullseye (11))
- Homebridge Homebridge v1.8.1
- Plugin v10.6.0
- Nodejs v20.9.0

**Comment by schwaggy** [comment]
I've stayed on 10.2.0 of the PI until @bwp91 can let us know plans for this issue. Personally, I don't have any other Govee devices and only purchased this model leak sensor because of this plug-in. It sure would be nice to have this issue fixed so I can update and trust the system again! The native Govee app was always hit or miss and I'd prefer to not have to log in or rely on it.

**Comment by mynameisbobwehadababyitsaboy** [comment]
> I've stayed on 10.2.0 of the PI until @bwp91 can let us know plans for this issue. Personally, I don't have any other Govee devices and only purchased this model leak sensor because of this plug-in. It sure would be nice to have this issue fixed so I can update and trust the system again! The native Govee app was always hit or miss and I'd prefer to not have to log in or rely on it.

Good to know that 10.2.0 is working, I just rolled back and they are all back online! Good enough fix for me for now. :)

**Comment by bwp91** [comment]
Any improvement with the beta version of the plugin?

**Comment by schwaggy** [comment]
> Any improvement with the beta version of the plugin?

No, no change.

**Comment by bwp91** [comment]
i was looking at the differences between v10.2.0 and v10.3.0 and could not see anything really that could explain why this lost functionality.

Is is absolutely definitely 💯 the case that 10.2.0 works and 10.3.0 doesn't?

**Comment by schwaggy** [comment]
Yeah, just tested again. Battery status shows 0% for all sensors, and one other difference I see is running 10.2.0, the ID of the sensors is in this format: [42F52038CBF60018C3] but 10.3.x and higher is: [42:F5:20:38:CB:F6:00:18:C3]
No idea if that makes any difference but just noticed it. Thanks!

**Comment by bwp91** [comment]
@schwaggy thanks, any luck with 10.6.1-beta.1?

**Comment by schwaggy** [comment]
> @schwaggy thanks, any luck with 10.6.1-beta.1?

YES! Working again!!! Logs are going crazy tho - is that just because of beta status?
Thank You!

**Comment by RJ1002** [comment]
> @schwaggy thanks, any luck with 10.6.1-beta.1?

can also confirm that my water sensor started working on 10.6.1-beta.1

it might be something on my end but in Home Assistant it showing the Devices twice one has the water sensor the battery% and identify button and the other has nothing.

**Comment by lsliddar** [comment]
Just updated to 10.7 -- sensors seem to show up properly within Homebridge (battery %, etc) when I login but not seeing anything in Apple Home app yet. Anyone else?

**Comment by schwaggy** [comment]
> Just updated to 10.7 -- sensors seem to show up properly within Homebridge (battery %, etc) when I login but not seeing anything in Apple Home app yet. Anyone else?

You will only see them listed as Alarm Sensors (top right) in the room they are assigned to. You will not see them as buttons or favorites.

**Comment by lsliddar** [comment]
> > Just updated to 10.7 -- sensors seem to show up properly within Homebridge (battery %, etc) when I login but not seeing anything in Apple Home app yet. Anyone else?
> 
> You will only see them listed as Alarm Sensors (top right) in the room they are assigned to. You will not see them as buttons or favorites.

Ugh... I'm an idiot. Thanks - I had added them to 'Homebridge' as the Room vs the actual room they were located. Showing up accurately with battery % detail as well. Thanks!

**Comment by hunterzero99** [comment]
I've been messing around with my water sensors and bringing them into homebridge for about a week now. I think the issue comes from having the Govee app installed/running in the background on your Govee account the same time as the homebridge app. The Govee app seems to use a ton of API calls, I pretty much always get locked out for up to 24 hours just when adding my 3 leak sensors. 

I forgot to Disable the app on my phone yesterday and when I went to test the sensors just bow they weren't working again, after over 24 hours of success with only homebridge. 

I am thinking a possible solution may be to create a 2nd Govee account with a different email address and share the devices with that account.  Then you can maybe use that one only for homebridge, while logging the app into your regular account. This would let you clear the sensor leak status too, which seems to be something I have to do.

**Comment by hunterzero99** [comment]
I've been messing around with my water sensors and bringing them into homebridge for about a week now. I think the issue comes from having the Govee app installed/running in the background on your Govee account at the same time as the homebridge app. The Govee app seems to use a ton of API calls, I pretty much always get locked out for up to 24 hours just when adding my 3 leak sensors. 

I forgot to Disable the app on my phone yesterday and when I went to test the sensors just now they weren't working again, after over 24 hours of success with only homebridge.  

The real downside is that I have to use the app to clear the leak status from any devices after it's triggered. Otherwise they stay in leak for me.

**Comment by brylee123** [comment]
Has anyone encountered this issue again lately? I am running on the most up to date version of the plugin. I see that hunterzero99 mentioned disabling the app, but that's not a possibility on my end because I have other devices that I use the app for.
